watertap/unit_models/translators/translator_asm1_adm1.py

```diff
@@ -0,0 +1,691 @@
+            self.flowsheet().time,
+            initialize=self.properties_in[0].conc_mass_comp["X_BA"],
+            units=pyunits.kg / pyunits.m**3,
+            domain=PositiveReals,
+            doc="Active autotrophic biomass remaining",
+        )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="COD demand minus active autotrophic biomass",
+        )
+        def CODd_step4(blk, t):
+            return (
+                blk.inter_X_BA[t]
+                == blk.properties_in[t].conc_mass_comp["X_BA"] - blk.CODd4[t]
+            )
+
+        @self.Expression(self.flowsheet().time, doc="COD demand after X_BA")
+        def CODd5(blk, t):
+            return 0.5 * (
+                (blk.CODd4[t] - blk.properties_in[t].conc_mass_comp["X_BA"])
+                + sqrt(
+                    (blk.CODd4[t] - blk.properties_in[t].conc_mass_comp["X_BA"]) ** 2
+                    + eps
+                )
+            )
+
+        @self.Expression(self.flowsheet().time, doc="Soluble COD")
+        def CODs(blk, t):
+            return blk.properties_in[t].conc_mass_comp["S_I"] + blk.inter_S_S[t]
+
+        @self.Expression(self.flowsheet().time, doc="Particulate COD")
+        def CODp(blk, t):
+            return (
+                blk.properties_in[t].conc_mass_comp["X_I"]
+                + blk.inter_X_S[t]
+                + blk.inter_X_BH[t]
+                + blk.inter_X_BA[t]
+                + blk.properties_in[t].conc_mass_comp["X_P"]
+            )
+
+        @self.Expression(self.flowsheet().time, doc="Total COD")
+        def CODt(blk, t):
+            return blk.CODs[t] + blk.CODp[t]
+
+        @self.Expression(self.flowsheet().time, doc="Required soluble COD")
+        def ReqCODs(blk, t):
+            return (
+                blk.properties_in[t].conc_mass_comp["S_ND"]
+                / blk.config.reaction_package.N_aa
+                / mw_n
+            )
+
+        @self.Expression(self.flowsheet().time, doc="Amino acids mapping")
+        def saa_mapping(blk, t):
+            return Expr_if(
+                blk.inter_S_S[t] > blk.ReqCODs[t],
+                blk.ReqCODs[t],
+                blk.inter_S_S[t],
+            )
+
+        @self.Expression(self.flowsheet().time, doc="Monosaccharides mapping step A")
+        def ssu_mapping_A(blk, t):
+            return Expr_if(
+                blk.inter_S_S[t] > blk.ReqCODs[t],
+                blk.inter_S_S[t] - blk.ReqCODs[t],
+                1e-6 * pyunits.kg / pyunits.m**3,
+            )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Amino acids balance",
+        )
+        def Saa_balance(blk, t):
+            return blk.properties_out[t].conc_mass_comp["S_aa"] == blk.saa_mapping[t]
+
+        @self.Expression(
+            self.flowsheet().time,
+            doc="Monosacharides balance step A",
+        )
+        def Ssu_balance_A(blk, t):
+            return blk.ssu_mapping_A[t]
```

Comment:
Why does this not follow the convention of the rest of the balances (i.e. `blk.properties_out[t].conc_mass_comp["S_su"] == blk.ssu_mapping_A[t]`

Reply:
deleted